modules/rostests/kmtests/ntos_ps/PsQuota.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * PROJECT:     ReactOS kernel-mode tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Kernel mode tests for process quotas
+ * COPYRIGHT:   Copyright 2021 George Bișoc <george.bisoc@reactos.org>
+ */
+
+#include <kmt_test.h>
+
+START_TEST(PsQuota)
+{
+    NTSTATUS Status;
+    VM_COUNTERS VmCounters;
+    QUOTA_LIMITS QuotaLimits;
+    SIZE_T NonPagedUsage, PagedUsage;
+    PEPROCESS Process = PsGetCurrentProcess();
+
+    /* Guard the quota operations in a guarded region */
+    KeEnterGuardedRegion();
+
+    /* Report the current process' quota limits */
+    Status = ZwQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       &QuotaLimits,
+                                       sizeof(QuotaLimits),
+                                       NULL);
+    if (skip(NT_SUCCESS(Status), "Failed to query quota limits -- %lx\n", Status))
+    {
+        return;
+    }
+
+    trace("Process paged pool quota limit -- %lu\n", QuotaLimits.PagedPoolLimit);
+    trace("Process non paged pool quota limit -- %lu\n", QuotaLimits.NonPagedPoolLimit);
+    trace("Process page file quota limit -- %lu\n\n", QuotaLimits.PagefileLimit);
```

Comment:
You could also have the test check that `Usage < Limit`.
`Usage > 0` probably also makes sense.